lib/modules/manager/pip-compile/extract.ts

```diff
@@ -257,18 +254,18 @@ function extendWithIndirectDeps(
   lockedDeps: PackageDependency[],
 ): void {
   for (const lockedDep of lockedDeps) {
-    if (
-      !packageFileContent.deps.find(
-        (dep) =>
-          normalizeDepName(lockedDep.depName!) ===
-          normalizeDepName(dep.depName!),
-      )
-    ) {
+    if (!packageFileContent.deps.find((dep) => compareNames(dep, lockedDep))) {
       packageFileContent.deps.push(indirectDep(lockedDep));
     }
   }
 }
 
+function compareNames(a: PackageDependency, b: PackageDependency): boolean {
+  const aName = a.packageName ?? a.depName;
+  const bName = b.packageName ?? b.depName;
+  return aName === bName;
+}
+
```

Comment:
I don't like this, is there a more reliable way to compare names? Doing it inline didn't work.